pkg/credentialprovider/plugin/config_test.go

```diff
@@ -245,6 +248,19 @@ providers:
 				t.Error("expected error but got none")
 			}
 
+			if testcase.config != nil && authConfig != nil {
+				// append OS env vars and sort it
+				for i := range testcase.config.Providers {
+					appendSystemEnvVars(os.Environ(), &testcase.config.Providers[i])
```

Comment:
Seems odd to be appending system env vars into the test data, this is data we should be testing against not mutating.

Reply:
@andrewsykim Agreed to your point, but if I don't perform this step then all the test cases would fail since `authConfig` would have env vars present in the system appended and `testcase.config` wouldn't.

Reply:
I recommend following the approach taken by the client-go environment construction which stubs `environ `func() []string` for testing

Reply:
@liggitt Done. 

Also, I have appended the env vars during the execution + I have stubbed couple of function calls in `execPlugin` to facilitate unit testing